ush/soca/prep_ocean_obs.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python3
+
+from datetime import datetime, timedelta
+from gen_bufr2ioda_json import gen_bufr_json
+from logging import getLogger
+from multiprocessing import Process
+import os
+from soca import prep_ocean_obs_utils
+from typing import Dict
+from wxflow import (chdir,
+                    FileHandler,
+                    logit,
+                    save_as_yaml,
+                    Task,
+                    YAMLFile)
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class PrepOceanObs(Task):
+    """
+    Class for prepping obs for ocean analysis task
+    """
+
+    @logit(logger, name="PrepOceanObs")
+    def __init__(self, config: Dict) -> None:
+        """Constructor for ocean obs prep task
+        Parameters:
+        ------------
+        config: Dict
+            configuration, namely environment variables
+        Returns:
+        --------
+        None
+        """
+
+        logger.info("init")
+        super().__init__(config)
+
+        PDY = self.runtime_config['PDY']
+        cyc = self.runtime_config['cyc']
+        cdate = PDY + timedelta(hours=cyc)
+
+        self.runtime_config['cdate'] = cdate
+        window_begin_datetime = cdate - timedelta(hours=3)
```

Comment:
This should be customizable with something like "half of assim_freq"

Reply:
done